src/bitcoin/mod.rs

```diff
@@ -156,14 +166,17 @@ impl BitcoinInterface for sync::Arc<sync::Mutex<dyn BitcoinInterface + 'static>>
         self.lock().unwrap().received_coins(tip)
     }
 
-    fn confirmed_coins(&self, outpoints: &[bitcoin::OutPoint]) -> Vec<(bitcoin::OutPoint, i32)> {
+    fn confirmed_coins(
+        &self,
+        outpoints: &[bitcoin::OutPoint],
+    ) -> Vec<(bitcoin::OutPoint, i32, u32)> {
         self.lock().unwrap().confirmed_coins(outpoints)
     }
 
     fn spent_coins(
         &self,
         outpoints: &[bitcoin::OutPoint],
-    ) -> Vec<(bitcoin::OutPoint, bitcoin::Txid)> {
+    ) -> Vec<(bitcoin::OutPoint, bitcoin::Txid, u32)> {
```

Comment:
We don't, and returning `None` *is* the issue i'm talking about. :sweat_smile: 

In the very same commit you modify the calling code too. You can see it will store a dummy txid if we return `None` here. And we *will* return `None` most of the time!

Reply:
I thought my reply was on comment https://github.com/revault/minisafe/pull/21#discussion_r971790132. Sorry for the confusion. But the issue is the same, we can't assume spending transactions are confirmed.